addons/mail/static/src/core/common/thread.js

```diff
@@ -218,7 +218,7 @@ export class Thread extends Component {
                     return;
                 }
                 const el = this.messageRefs.get(
-                    this.props.thread.self_member_id.new_message_separator_ui - 1
+                    this.props.thread.channel.self_member_id.new_message_separator_ui - 1
```

Comment:
Even before this change, this is quite fragile, isn't it? It relies on the fact that `jumpToNewMessage` props is only passed for channel the user is actually a member of. It looks very odd to do such assumptions, especially since the `Thread` component handles jumping to messages for non channels. Shouldn't we guard this `useEffect`?